pricegraph/src/orderbook.rs

```diff
@@ -166,6 +166,14 @@ impl Orderbook {
         base: TokenId,
         quote: TokenId,
     ) -> TransitiveOrderbook {
+        if !self.is_token_pair_valid(TokenPair {
+            buy: base,
+            sell: quote,
+        }) || base == quote
```

Comment:
Without this check the new test passes but the fuzzer finds an instance where a transitive order ends up as `NaN`. I haven't saved the instance but if it's interesting it should be able to find it again.

Conceptually it doesn't make sense to request the transitive orderbook for the same token, does it? If it does then we might have to fix this in a different way.

Reply:
> Conceptually it doesn't make sense to request the transitive orderbook for the same token, does it?

Hmm, I can't think of a particular use case for it. I think its fine for now.